tests/python/test-parameter-server.py

```diff
@@ -9,360 +10,56 @@
-      <origin rpy=\"0 0 0\" xyz=\"0 0 0\"/>\
-      <geometry>\
-        <box size=\"0.05 0.05 0.05\"/>\
-      </geometry>\
-    </collision>\
-    <visual>\
-      <origin rpy=\"0 0 0\" xyz=\"0 0 0\"/>\
-      <geometry>\
-        <box size=\"0.05 0.05 0.05\"/>\
-      </geometry>\
-      <material name=\"red\"/>\
-    </visual>\
-    <inertial>\
-      <mass value=\"1e-5\"/>\
-      <origin rpy=\"0 0 0\" xyz=\"0 0 0\"/>\
-      <inertia ixx=\"1e-6\" ixy=\"0\" ixz=\"0\" iyy=\"1e-6\" iyz=\"0\" izz=\"1e-6\"/>\
-    </inertial>\
-  </link>\
-  <!-- generate an optical frame http://www.ros.org/reps/rep-0103.html#suffix-frames\
-      so that ros and opencv can operate on the camera frame correctly -->\
-  <joint name=\"camera_optical_joint\" type=\"fixed\">\
-    <!-- these values have to be these values otherwise the gazebo camera image\
-        won't be aligned properly with the frame it is supposedly originating from -->\
-    <origin rpy=\"-1.57079632679 0 -1.57079632679\" xyz=\"0 0 0\"/>\
-    <parent link=\"camera_link\"/>\
-    <child link=\"camera_link_optical\"/>\
-  </joint>\
-  <link name=\"camera_link_optical\">\
-  </link>\
-  <transmission name=\"tran1\">\
-    <type>transmission_interface/SimpleTransmission</type>\
-    <joint name=\"joint1\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-    </joint>\
-    <actuator name=\"motor1\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-      <mechanicalReduction>1</mechanicalReduction>\
-    </actuator>\
-  </transmission>\
-  <transmission name=\"tran2\">\
-    <type>transmission_interface/SimpleTransmission</type>\
-    <joint name=\"joint2\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-    </joint>\
-    <actuator name=\"motor2\">\
-      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>\
-      <mechanicalReduction>1</mechanicalReduction>\
-    </actuator>\
-  </transmission>\
-</robot>\
-"
+import sys
 
-# 
-model = pin.buildModelFromXML(urdf_rrbot_model_string)
+from os.path import dirname, join, abspath
 
-print("model.nq:"+str(model.nq))
-print(len(q))
-neutral_configuration_model=pin.neutral(model)
-#print(model)
-jointIds=[]
-reduced_model=pin.buildReducedModel(model, jointIds, q)
-print(neutral_configuration_model)
-print(len(q))
+class TestParameterServer(unittest.TestCase):
 
-data = model.createData()
-com = pin.centerOfMass(model, data, q)
-pin.updateFramePlacements(model, data)
-m = data.mass[0]
-param_server.setParameter("urdf_model",urdf_rrbot_model_string)
-model2_string=param_server.getParameter("urdf_model")
-print(len(model2_string))
+    def test_set_parameter(self):
+        # Read talos model
+        path = join(dirname(dirname(abspath(__file__))),
+                    'models', 'others', 'python')
+        sys.path.append(path)
 
+        from example_robot_data.path import EXAMPLE_ROBOT_DATA_MODEL_DIR
```

Comment:
This adds a dependency to an external package. This test only checks the set/get parameter works. The URDF is parsed but I don't think this is useful so I guess even a string with rubbish content would work. Am I right ?
Or there is something you intend to check that I didn't catch.

Reply:
Well on this specific test you are right. But this package needs a better test coverage which will imply to read a real model.
So after thinking over it, I think that this specific dependency is fine.